modules/overlays/trilby-cli.nix

```diff
@@ -1,8 +1,14 @@
 { inputs, lib, trilby, ... }:
 
-self: super:
-(import ../../trilby-cli/overlay.nix { inherit inputs lib; }) self super
-  //
+final: prev:
+let
+  flake = lib.loadFlake {
+    inherit (prev) system;
+    src = ../../trilby-cli;
+  };
+  overlay = flake.defaultNix.outputs.overlays.default;
+  pkgs = prev.extend overlay;
+in
 {
-  trilby-cli = self.haskellPackages.trilby-cli.bin;
+  trilby-cli = pkgs.haskellPackages.trilby-cli.bin;
```

Comment:
Wouldn't it make more sense to use `flake.defaultNix.outputs.packages.${system}.default` here? Instead of building an entire package set (which wouldn't even have all the overlays).

Reply:
I specifically want to avoid the trilby-cli specific nixpkgs in favour of the ones used in trilby proper.

The trilby-cli nixpkgs has nixpkgs only for shell, it might as well not have any packages set.

Reply:
I'm mostly concerned that you can't use `final.extend overlay` because it'd be circular (I think? Maybe nix could be smart enough...). The `prev.extend ...` makes the overlays available here depend on the order imposed by filenames in `modules/overlays`. For instance, if I'd need to add a overlay to fix Haskell on some exotic architecture, it'd be a gamble whether `trilby-cli` would be built with it. On the other hand, my suggestion above would have had the same issue. I say, we shelve that concern for now.